In [1]:
from architecture import INN as INN
from architecture import MNIST_autoencoder as mnist
from architecture import CIFAR_autoencoder as cifar
from architecture import CelebA_autoencoder as celeba
from functionalities import parameter_counter as count
from torch import nn

In [2]:
mnist_model = mnist.mnist_autoencoder(1)
print('min:', count.count_para(mnist_model))

mnist_model = mnist.mnist_autoencoder(28*28)
print('max:', count.count_para(mnist_model))

2934904
5861

min: 485137
max: 586144


In [3]:
cifar_model = cifar.cifar_autoencoder(1)
print('min:', count.count_para(cifar_model))

cifar_model = cifar.cifar_autoencoder(3*32*32)
print('max:', count.count_para(cifar_model))

min: 2904996
max: 31210403


In [4]:
celeba_model = celeba.celeba_autoencoder(1)
print('min:', count.count_para(celeba_model))

celeba_model = celeba.celeba_autoencoder(3*218*178)
print('max:', count.count_para(celeba_model))

min: 45171524


RuntimeError: $ Torch: not enough memory: you tried to allocate 5GB. Buy new RAM! at /opt/conda/conda-bld/pytorch-cpu_1532576596369/work/aten/src/TH/THGeneral.cpp:204

In [2]:
mnist_inn_model = INN.mnist_inn_com()
print(count.count_para(mnist_inn_model))

# 471576

Node r1 has following input dimensions:
	 Output #0 of node input: (1, 28, 28)

Node conv1 has following input dimensions:
	 Output #0 of node r1: (4, 14, 14)

Node conv2 has following input dimensions:
	 Output #0 of node conv1: (4, 14, 14)

Node conv3 has following input dimensions:
	 Output #0 of node conv2: (4, 14, 14)

Node r2 has following input dimensions:
	 Output #0 of node conv3: (4, 14, 14)

Node output has following input dimensions:
	 Output #0 of node r2: (1, 28, 28)

242712


In [2]:
cifar_inn_model = INN.cifar_inn_com(1)
print('min:', count.count_para(cifar_inn_model))

cifar_inn_model = INN.cifar_inn_com(3*32*32)
print('max:', count.count_para(cifar_inn_model))

Node r1 has following input dimensions:
	 Output #0 of node input: (3, 32, 32)

Node conv11 has following input dimensions:
	 Output #0 of node r1: (12, 16, 16)

Node conv12 has following input dimensions:
	 Output #0 of node conv11: (12, 16, 16)

Node conv13 has following input dimensions:
	 Output #0 of node conv12: (12, 16, 16)

Node r2 has following input dimensions:
	 Output #0 of node conv13: (12, 16, 16)

Node conv21 has following input dimensions:
	 Output #0 of node r2: (48, 8, 8)

Node conv22 has following input dimensions:
	 Output #0 of node conv21: (48, 8, 8)

Node conv23 has following input dimensions:
	 Output #0 of node conv22: (48, 8, 8)

Node r3 has following input dimensions:
	 Output #0 of node conv23: (48, 8, 8)

internal_size: 1
internal_size: 1
internal_size: 1
internal_size: 1
Node fc has following input dimensions:
	 Output #0 of node r3: (3072,)

Node r4 has following input dimensions:
	 Output #0 of node fc: (3072,)

Node r5 has following input dimensions:
	 

In [2]:
celeba_inn_model = INN.celeba_inn_com(1)
print('min:', count.count_para(celeba_inn_model))

celeba_inn_model = INN.celeba_inn_com(3*32*32)
print('max:', count.count_para(celeba_inn_model))

Node r1 has following input dimensions:
	 Output #0 of node input: (3, 218, 178)

Node conv11 has following input dimensions:
	 Output #0 of node r1: (12, 109, 89)

Node conv12 has following input dimensions:
	 Output #0 of node conv11: (12, 109, 89)

Node conv13 has following input dimensions:
	 Output #0 of node conv12: (12, 109, 89)

Node r2 has following input dimensions:
	 Output #0 of node conv13: (12, 109, 89)



AssertionError: Uneven input dimensions

# Notiz

MNIST INN Autoencoder: 471576 Parameter 

MNIST Autoencoder: 485137 Parameter (Minimum)
                   586144 Parameter (Maximum)

CIFAR INN Autoencoder: 118558416 Parameter

CIFAR Autoencoder: 2904996 Parameter (Minimum)
                   31210403 Parameter (Maximum)
                   


In [12]:
model = autoencoder(60)
print(count.count_para(model))

286047


In [ ]:
117664104 (fully)
42142056 (small)
4393324 (min inn)
42142056 (max inn)
2904996 (min)
31210403 (max)